web/yo/app/scripts/controllers/gene.js

```diff
@@ -3009,6 +3035,184 @@ angular.module('oncokbApp')
+                    drugs.push(drugVal);
+                })
+                return drugs.filter(function (drug) {
+                    if (!drug.synonyms)
+                        return drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1
+                    else return (drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1) || (drug.synonyms.join(',').toLowerCase().indexOf($query.toLowerCase()) != -1);
+                })
+            });
+        }
+
+        $scope.addDruginElement = function (tag, index) {
+            therapyUuid[index].push(tag.uuid);
+            addTherapy(index);
+        }
+
+        $scope.removeDruginElement = function (tag, index) {
+            var id = therapyUuid[index].indexOf(tag.uuid);
+            if (id > -1) {
+                therapyUuid[index].splice(id, 1);
+            }
+            addTherapy(index);
+        }
+
+
+        function addTherapy(index) {
+            if ($scope.therapy.length === index + 1) {
+                $scope.therapy.push([]);
+                therapyUuid.push([]);
+            }
+            valiElement(index);
+
+        }
+
+        function valiElement(index) {
+            $scope.addTherapyError = false;
+            var therapyVali = [];
+            _.each(therapyUuid, function (element) {
+                element = element.sort();
+                element = element.join(' ').trim();
+                if (element !== '')
+                    therapyVali.push(element);
+            });
+            for (var i = 0; i < therapyVali.length; i++) {
+                for (var j = i + 1; j <= therapyVali.length; j++) {
+                    if (therapyVali[i] === therapyVali[j]) {
+                        $scope.therapyErrorMessage = "Same Elements. Please check and save again.";
+                        $scope.addTherapyError = true;
+                    }
+                }
+            }
+        }
+
+        $scope.deleteTherapy = function (index) {
+            if (index > 0) {
+                $scope.therapy.splice(index, 1);
+                therapyUuid.splice(index, 1);
+            }
+            valiElement(index);
+        }
+
+        function isValidTreatment(indices, newTreatmentName) {
+            var isValid = true;
+            _.find(data.tiRef.treatments, function (treatment) {
+                if (treatment.name === newTreatmentName) {
+                    isValid = false;
+                    return true;
+                }
+            });
+            if (!isValid) {
+                $scope.therapyErrorMessage = "Same therapy exists.";
+                $scope.addTherapyError = true;
+            }
+            return isValid;
+        }
+
+        $scope.save = function () {
+            for (; therapyUuid[therapyUuid.length - 1] == '';) {
+                $scope.therapy.splice($scope.therapy.length - 1, 1);
+                therapyUuid.splice(therapyUuid.length - 1, 1);
+            }
+            ;
```

Comment:
Generally, there is only one line which is blank when user clicks Save button. It may be possible that user leave the last few lines contentless. Codes in this part are for deleting these blank lines.

Reply:
My question is why the semicolon here?

Reply:
Sorry, it has been deleted.